test.lua

```diff
@@ -1095,15 +1095,18 @@ function nntest.SpatialConvolution()
 end
 
 function nntest.SpatialConvolutionMM()
-   local from = math.random(1,5)
+   local from = math.random(2,5)
    local to = math.random(1,5)
-   local ki = math.random(1,3)
-   local kj = math.random(1,3)
+   local ki = math.random(1,5)
+   local kj = math.random(1,5)
+   local di =  math.random(1,4)
+   local dj =  math.random(1,4)
+   local padding = math.random(0,2)
    local outi = math.random(5,9)
    local outj = math.random(5,9)
-   local ini = outi-1+ki
-   local inj = outj-1+kj
-   local module = nn.SpatialConvolutionMM(from, to, ki, kj)
+   local ini = (outi-1)*si+ki-padding*2
```

Comment:
si and sj are not defined, you must mean di and dj.
On that note, did you unit test this? as your PR as such wont run the unit test because si/sj are not defined.